modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/source/SourceMatcher.java

```diff
@@ -179,7 +179,8 @@ private Optional<MatchResult> matchWithFieldSpecificMatcher(String fieldName, Li
 
     // Checks for scenarios when source is stored exactly and therefore can be compared without special logic.
     private boolean sourceMatchesExactly(MappingTransforms.FieldMapping mapping, List<Object> expectedValues) {
-        return mapping.parentMappingParameters().stream().anyMatch(m -> m.getOrDefault("enabled", "true").equals("false"));
+        return mapping.parentMappingParameters().stream().anyMatch(m -> m.getOrDefault("enabled", "true").equals("false"))
+            || mapping.mappingParameters().getOrDefault("synthetic_source_keep", "none").equals("all");
```

Comment:
@lkts any hints on how to cover `arrays` too?

Reply:
Sounds good, there should be no complaints if we accidentally produce the same source.. 

But isn't this the case today, based on the code you call out above?

Reply:
Yes, you can just remove `sourceMatchesExactly`.

Reply:
Sorry I misunderstood your suggestion. I think the current form is correct, the test should report an error if the source is expected to be an exact copy but it's not - even if relaxed matching succeeds.